awscli/customizations/s3/filegenerator.py

```diff
@@ -304,10 +302,8 @@ def _list_single_object(self, s3_path):
         # IAM policies with the smallest set of permissions needed) and
         # instead use a HeadObject request.
         bucket, key = find_bucket_key(s3_path)
-        operation = self._service.get_operation('HeadObject')
         try:
-            response = operation.call(
-                self._endpoint, bucket=bucket, key=key)[1]
+            response = self._client.head_object(Bucket=bucket, Key=key)
         except ClientError as e:
             # We want to try to give a more helpful error message.
             # This is what the customer is going to see so we want to
```

Comment:
At the same time, the former is the same error message you get when you call `s3api head-object` so the consistency is nice.  I think the 404 vs. NoSuchKey is not that big of a deal, I'd think most people would understand a 404.  However, the latter message has the key name which I think is really helpful.  What about keeping the code the same, but just changing the error message to include the key name?